spring-integration-sftp/src/main/java/org/springframework/integration/sftp/outbound/SftpMessageHandler.java

```diff
@@ -66,6 +66,14 @@ public SftpMessageHandler(SessionFactory<LsEntry> sessionFactory) {
 		this(new SftpRemoteFileTemplate(sessionFactory));
 	}
 
+	public SftpMessageHandler(RemoteFileTemplate<LsEntry> remoteFileTemplate) {
```

Comment:
We can't introduce these new ctors based on the super `RemoteFileTemplate`.
The specs must be restricted only to their specific `SftpRemoteFileTemplate` or `FtpRemoteFileTemplate`.

It's kinda breaking change, but we can introduce new ctors an factory methods with deprecation of those based on just a `RemoteFileTemplate`

Reply:
so changing this to

```
SftpMessageHandlerSpec(RemoteFileTemplate<ChannelSftp.LsEntry> remoteFileTemplate) {
		this.target = new SftpMessageHandler(remoteFileTemplate.getSessionFactory());
	}

	SftpMessageHandlerSpec(RemoteFileTemplate<ChannelSftp.LsEntry> remoteFileTemplate, FileExistsMode fileExistsMode) {
		this.target = new SftpMessageHandler(new SftpRemoteFileTemplate(remoteFileTemplate.getSessionFactory()), fileExistsMode);
	}
``` 

and removing ctors makes sense for you?

Reply:
should be ok now, thanks for the feeback!